[Misc] Replace os environ to monkeypatch in test suite

t-sibiraj opened the pull request:
Convert all os.environ(xxx) to monkeypatch.setenv in test suite

Submitting pull request again because I forget to sign my original pull request

FIXES [#14499](https://github.com/vllm-project/vllm/issues/14499)

Co-authored-by: Aaron Pham <contact@aarnphm.xyz>
Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

t-sibiraj committed:
Replace os.environ with monkeypatch in test suite

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
Replace os.environ with monkeypatch in test suite

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

DarkLight1337 commented on tests/basic_correctness/test_cumem.py:
```diff
@@ -159,4 +159,4 @@ def test_end_to_end(model: str, use_v1: bool):
     assert output[0].outputs[0].text == output2[0].outputs[0].text
 
-    del os.environ["VLLM_USE_V1"]
+    monkeypatch.delenv("VLLM_USE_V1", raising=False)
```
```suggestion
    monkeypatch.delenv("VLLM_USE_V1", raising=True)
```

Since `del` also raises an error if the key doesn't exist.

DarkLight1337 commented on tests/test_utils.py:
```diff
@@ -112,16 +112,16 @@ def dummy(*, old_arg: object = None, new_arg: object = None):
             with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s3:
                 s3.bind(("localhost", get_open_port()))
-    os.environ.pop("VLLM_PORT")
+    monkeypatch.delenv("VLLM_PORT")
```
```suggestion
    monkeypatch.delenv("VLLM_PORT", raising=True)
```

t-sibiraj committed:
Update tests/basic_correctness/test_cumem.py

Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>
Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
fix: Use `raising=True` in `monkeypatch.delenv` for safer environment variable removal

Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>
Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

robertgshaw2-redhat commented on tests/models/decoder_only/language/test_gptq_marlin.py:
```diff
@@ -18,7 +18,12 @@
 from ...utils import check_logprobs_close
 
-os.environ["TOKENIZERS_PARALLELISM"] = "true"
+@pytest.fixture(scope="module", autouse=True)
```
you cant pass monkeypatch to a module level fixture

robertgshaw2-redhat commented on tests/models/decoder_only/vision_language/test_models.py:
```diff
@@ -29,8 +29,13 @@
 # FIXME (mattwong, gshtrasb, hongxiayan)
-if current_platform.is_rocm():
-    os.environ["VLLM_USE_TRITON_FLASH_ATTN"] = "0"
+@pytest.fixture(scope="module", autouse=True)
```
ditto

robertgshaw2-redhat commented:
I would suggest installing the pre-commit which will automatically sign off for you. Run `./format.sh` locally and it will give you instructions

t-sibiraj committed:
doing a signed commit

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

DarkLight1337 commented on tests/models/decoder_only/vision_language/test_models.py:
```diff
@@ -29,6 +29,7 @@
 # ROCm Triton FA can run into shared memory issues with these models,
 # use other backends in the meantime
 # FIXME (mattwong, gshtrasb, hongxiayan)
+
```
```suggestion
```

Remove unnecessary change

t-sibiraj committed:
remove unnecessary spacing

Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>
Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
Intentionally added and removed a space to test this commit, so I could push the changes and pass the pre-commit check

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

aarnphm commented on tests/basic_correctness/test_basic_correctness.py:
```diff
@@ -47,6 +47,7 @@ def test_vllm_gc_ed():
 @pytest.mark.parametrize("max_tokens", [5])
 @pytest.mark.parametrize("enforce_eager", [False])
 def test_models(
+    monkeypatch,
```
s/monkeypatch/monkeypatch: pytest.MonkeyPatch

t-sibiraj replied:
will reference it explicitly

t-sibiraj committed:
add type annotations for monkeypatch in the function  header and add import statement for pytest in test/compile/utils.py

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
changed the order of the parameter in function header

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
Resolved merge conflict

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj commented:
I will add the fix for the check_full_graph_support function and push the changes again

t-sibiraj committed:
remove unnessary arguements passed to function

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
add necesssary arguements to the check_full_graph_support function

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
reverted back to use os.environ since pyttest cannot be passed to a module level fixture

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

DarkLight1337 commented:
Please fix the pre-commit errors

t-sibiraj committed:
pre commit changes

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

aarnphm commented on tests/kernels/test_awq.py:
```diff
@@ -11,8 +9,8 @@
                     reason="AWQ is not supported on this GPU type.")
-def test_awq_dequantize_opcheck():
-    os.environ["VLLM_USE_TRITON_AWQ"] = "0"
+def test_awq_dequantize_opcheck(monkeypatch):
```
```suggestion
def test_awq_dequantize_opcheck(monkeypatch: pytest.MonkeyPatch):
```

aarnphm commented on tests/kernels/test_awq.py:
```diff
@@ -29,8 +27,8 @@ def test_awq_dequantize_opcheck():
                     reason="AWQ is not supported on this GPU type.")
-def test_awq_gemm_opcheck():
-    os.environ["VLLM_USE_TRITON_AWQ"] = "0"
+def test_awq_gemm_opcheck(monkeypatch):
```
```suggestion
def test_awq_gemm_opcheck(monkeypatch: pytest.MonkeyPatch):
```

t-sibiraj replied:
In my latest commit I have added the type annotations in all the function headers for the monkeypatch parameter

t-sibiraj committed:
add missing monkeypatch type annotations in function header

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
To resolve merge conflict
Merge remote-tracking branch 'upstream/main' into replace-os-environ-monkeypatch

aarnphm commented on tests/basic_correctness/test_basic_correctness.py:
```diff
@@ -118,11 +120,11 @@ def test_models_distributed(
         # test Ray Compiled Graph
-        os.environ['VLLM_USE_RAY_SPMD_WORKER'] = "1"
-        os.environ['VLLM_USE_RAY_COMPILED_DAG'] = "1"
+        monkeypatch.setenv("VLLM_USE_RAY_SPMD_WORKER", "1")
```
actually, one final thing, can you move this within `monkeypatch.context` instead of just setenv here?

https://docs.pytest.org/en/stable/reference/reference.html#pytest.MonkeyPatch.context

We probably don't want to have any side-effects here.

aarnphm reviewed (approved):
one final thing, then GTG

cc @DarkLight1337 @robertgshaw2-redhat when you guys have bandwidth.

t-sibiraj committed:
move monkeypatch.setenv within monkeypatch.context

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj committed:
Merge branch 'main' into replace-os-environ-monkeypatch

aarnphm commented on tests/basic_correctness/test_basic_correctness.py:
```diff
@@ -116,13 +118,15 @@ def test_models_distributed(
-        # test Ray Compiled Graph
-        os.environ['VLLM_USE_RAY_SPMD_WORKER'] = "1"
-        os.environ['VLLM_USE_RAY_COMPILED_DAG'] = "1"
+    with monkeypatch.context() as monkeypatch_context:
```
You have to add the whole block within this context manager...

aarnphm commented:
@t-sibiraj you can apply this patch befor 4932bcd

[14516.patch](https://github.com/user-attachments/files/19261021/14516.patch)

t-sibiraj committed:
Apply patch before 4932bcd

Signed-off-by: sibi <85477603+t-sibiraj@users.noreply.github.com>

t-sibiraj commented:
> @t-sibiraj you can apply this patch befor [4932bcd](https://github.com/vllm-project/vllm/commit/4932bcd8de27f2a3b615434ff972e0b16252bedf)
> 
> [14516.patch](https://github.com/user-attachments/files/19261021/14516.patch)

I will apply the patch

aarnphm commented:
hmm, can you add me as collaborator to your branch? I can help with expedite this PR and fix CI error :)

t-sibiraj commented:
> hmm, can you add me as collaborator to your branch? I can help with expedite this PR and fix CI error :)

Thank you I really appreciate your help. I have sent you an invite.

aarnphm committed:
fix: correct tests

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

aarnphm committed:
fix: correct types

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

aarnphm committed:
fix: ignore otel imports mypy

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

aarnphm committed:
fix: types

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

aarnphm committed:
fix: distributed tests with dispatch

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

aarnphm committed:
chore: fix tests

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

aarnphm commented:
@t-sibiraj can you add the following to the PR description?

```text
Co-authored-by: Aaron Pham <contact@aarnphm.xyz>
Signed-off-by: Aaron Pham <contact@aarnphm.xyz>
```

Thanks

aarnphm committed:
fix: precommit error

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

DarkLight1337 commented:
Done, and thanks for fixing this!

ProExpertProg commented:
This PR broke the PyTorch Fullgraph tests:
```
[2025-03-16T17:00:51Z] Running 2 items in this shard: tests/compile/test_full_graph.py::test_full_graph[model_info0-2], tests/compile/test_full_graph.py::test_full_graph[model_info0-3]
--
  | [2025-03-16T17:00:51Z]
  | [2025-03-16T17:00:51Z] compile/test_full_graph.py::test_full_graph[model_info0-2] SKIPPED (...)
  | [2025-03-16T17:00:51Z] compile/test_full_graph.py::test_full_graph[model_info0-3] SKIPPED (...)

```
The fixture cannot be used in this simple way to generate an array of parameters
